spec/support/corpus.rb

```diff
@@ -134,15 +138,33 @@ def install_mutant
         system(%w[bundle])
       end
 
-      # Not in the docs. Number from chatting with their support.
-      CIRCLE_CI_CONTAINER_PROCESSES = 2
+      # The effective ruby file paths
+      #
+      # @return [Array<Pathname>]
+      def effective_ruby_paths
+        paths = Pathname
+          .glob(repo_path.join('**/*.rb'))
+          .sort_by(&:size)
+          .reverse
+
+        paths - excluded_paths
+      end
+
+      # The excluded file paths
+      #
+      # @return [Array<Pathname>]
+      def excluded_paths
+        exclude.flat_map do |pattern|
+          Pathname.glob(repo_path.join(pattern))
+        end
+      end
 
       # Number of parallel processes to use
       #
       # @return [Fixnum]
       def parallel_processes
-        if ENV['CI']
-          Mutant::Config::DEFAULT.jobs
+        if ENV.key?('CI')
```

Comment:
> I'd be interested in hearing why mutant shouldn't be CI aware. I

Mutant should not read globals. Unless being told so or making this behavor (de-)configurable.

Before #502 mutant was setting very restrictive resource consumption details on detecting the `CI` environment variable. Which is a far to broad decision, as the previous resource consumption decision was based on one possible CI environment (CircleCI).

There is no way such a hard - coded default could possibly be correct for all possible CI environments. So its better to drop it.

Context for CI resource consumption problems: Docker and friends do not provide a way to introspect the *real* amount of threads allocated per container. `/proc/cpuinfo` reflect the hosts CPU without restrictions applied.

So mutant on CircleCI used to spawn 32 threads which quickly ran into resource barriers causing the build to be terminated by their watchdogs.

Also the previous hard limit on jobs under CI detection was set to `4` which depending on domain can be FAR too much or FAR to less, depending on domain under test.

So setting concurrency on CI is really the job of the dev, nothing mutant should do.

Same for everything else that should be environment aware.